### ISSUE TITLE
[Aio] Perform graceful shutdown for poller completion queue

### ISSUE THREAD

**Issue by lidizheng** [grpc/grpc#22365]
In https://github.com/grpc/grpc/pull/22311, the shutdown process of completion queue deadlocks. The SHUTDOWN event was successfully propagated to the Cython space, but the polling thread failed to re-acquire the GIL and resulted in a deadlock. This only happens when the completion queue is shutting down during the interpreter close.

**Comment by lidizheng** [comment]
Fixed by #22758

**Comment by colonelpanic8** [comment]
This is still a problem

**Comment by colonelpanic8** [comment]
Seems to be mostly fixed by makign sure that grpc is initialized early in the program and from the main thread:

```
    from grpc.experimental.aio import init_grpc_aio

    init_grpc_aio()
```

**Comment by sergiitk** [comment]
> Seems to be mostly fixed by makign sure that grpc is initialized early in the program and from the main thread:
> 
> ```
>     from grpc.experimental.aio import init_grpc_aio
> 
>     init_grpc_aio()
> ```

To clarify, the reason why calling `init_grpc_aio()`  manually worked, is that global_completion_queue's ref is incremented without any actual grpcio object using it, so the shutdown is prevented by off-by-one ref count. Normally the ref count only goes up when we initialize gRPC classes that actually use the queue, and decrement when these classes are deallocated.

Thus, when the app is exiting, the `global_completion_queue`'s ref count is still positive, which prevents the proper shutdown (`_actual_aio_shutdown`) from being called. While prevents some shutdown-associated errors (like the deadlock described here, as well as #38679, #33342, #36655, #39520), this approach has a major drawback.

This approach results in a memory leak when the app is done with gRPC calls, and proceeds to do other work: the `global_completion_queue` and the [grpc C core library](https://grpc.github.io/grpc/core/grpc_8h.html#a35f55253e80714c17f4f3a0657e06f1b) should been shut down because no gRPC objects are using them anymore, but the incorrect refcount keeps them up.

**Comment by sergiitk** [comment]
#40447 fixes these shutdown-associated errors, so `init_grpc_aio()` workaround should not be needed anymore. The fix will be included in v1.75.1 patch release.

I'm planning to mark `init_grpc_aio()` as deprecated some time soon, and eventually remove it from the public API surface.

**Comment by sergiitk** [comment]
[v1.75.1](https://github.com/grpc/grpc/releases/tag/v1.75.1) release contains the fix for this issue. Please post if this is still a problem.
